src/account.ts

```diff
@@ -824,109 +811,139 @@ export default class EthereumAccount {
     }
 
-    this.watcher = timer
+    return timer
   }
 
-  public claimIfProfitable (requireSettling = false): Promise<void> {
-    return this.claimTransaction.runExclusive(async () => {
-      try {
-        const claim = this.account.bestIncomingClaim
-        if (!claim) return
+  private claimIfProfitable (requireSettling = false): Promise<void> {
+    return this.queue.synchronize(TaskPriority.ClaimTx, async () => {
+      const claim = this.account.bestIncomingClaim
+      if (!claim) {
+        return
+      }
 
-        const channel = await this.updateChannel(claim.channelId)
-        if (!channel) {
-          return this.master._log.trace(`Cannot claim channel ${claim.channelId} with ${this.account.accountName}: linked channel doesn't exist or is settled`)
-        }
+      const channel = await this.fetchChannel(claim.channelId)
+      if (!channel) {
+        return this.master._log.trace(`Cannot claim channel ${claim.channelId} with ${this.account.accountName}: linked channel doesn't exist or is settled`)
+      }
 
-        if (requireSettling && !isSettling(channel)) {
-          return this.master._log.trace(`Cannot claim channel ${claim.channelId} with ${this.account.accountName}: channel is not settling`)
-        }
+      if (requireSettling && !isSettling(channel)) {
+        return this.master._log.trace(`Cannot claim channel ${claim.channelId} with ${this.account.accountName}: channel is not settling`)
+      }
 
-        this.master._log.trace(`Attempting to claim channel ${claim.channelId} for ${format(claim.value, Unit.Wei)}`)
+      this.master._log.trace(`Attempting to claim channel ${claim.channelId} for ${format(claim.value, Unit.Wei)}`)
 
-        const contract = await getContract(this.master._web3)
-        const txObj = contract.methods.claim(claim.channelId, claim.value, claim.signature)
-        const tx = await generateTx({
-          txObj,
-          from: channel.receiver,
-          web3: this.master._web3
-        })
+      const txObj = this.master._contract!.methods.claim(claim.channelId, claim.value, claim.signature)
+      const tx = await generateTx({
+        txObj,
+        from: channel.receiver,
+        web3: this.master._web3
+      })
 
-        // Check to verify it's profitable first
-        // This fee should already be accounted for in balance as apart of the initiation fee
-        const txFee = new BigNumber(tx.gasPrice).times(tx.gas)
-        if (txFee.gte(claim.value)) {
-          return this.master._log.trace(`Not profitable to claim incoming settling channel ${claim.channelId} with ${this.account.accountName}: ` +
-            `fee of ${format(txFee, Unit.Wei)} is greater than value of ${format(claim.value, Unit.Wei)}`)
-        }
+      // Check to verify it's profitable first
+      // This fee should already be accounted for in balance as a part of the initiation fee
+      const txFee = new BigNumber(tx.gasPrice).times(tx.gas)
+      if (txFee.gte(claim.value)) {
```

Comment:
If it's 1 Wei more, should it still claim it?

Reply:
Practically speaking, yeah, a net gain of 1 wei isn't very useful -- and you'd probably rather keep the channel open to save on fees the connector charges you if you want to use it in the future (deposits to existing channels are cheaper than opening new channels).

I would prefer not to add more config options (e.g. `minAmountToClaim`), but it's probably an argument for adding an admin API to close channels, so clients (or servers) can have their own logic for when to do so. That way it'd also be easier to expose to moneyd-gui for manually closing channels. (Do you have any ideas on that?)